lib/configuration.dart

```diff
@@ -0,0 +1,93 @@
+import 'constants.dart';
+import 'tracking_options.dart';
+import 'default_tracking.dart';
+
+/// Configuration for Amplitude instance.
+/// 
+/// Before initialzing Amplitude instance, create a Configuration instance
+/// with your desired configuration and pass it to the Amplitude instance.
+/// Notice that only `optOut` and be changed later by calling `setOptOut()`.
+/// Other configurations can *NOT* be changed after initialzing Amplitude instance.
+class Configuration {
+  String apiKey;
+  int flushQueueSize;
+  int flushIntervalMillis;
+  late String instanceName;
+  bool optOut;
+  LogLevel logLevel;
+  int? minIdLength;
+  String? partnerId;
+  int flushMaxRetries;
+  bool useBatch;
+  ServerZone serverZone;
+  String? serverUrl;
+  int minTimeBetweenSessionsMillis;
+  DefaultTrackingOptions defaultTracking;
+  TrackingOptions trackingOptions;
+  // Mobile (iOS and Android) specific
+  bool enableCoppaControl;
+  bool flushEventsOnClose;
+  int identifyBatchIntervalMillis;
+  bool migrateLegacyData;
+  // Android specific
+  bool locationListening;
```

Comment:
Is locationListening supported in Kotlin? Should this be changed to match new config options?

Reply:
Yes, it is supported. Please see [docs](https://www.docs.developers.amplitude.com/data/sdks/android-kotlin/#location-tracking). Also I tested it locally
<img width="547" alt="image" src="https://github.com/amplitude/Amplitude-Flutter/assets/31029607/deefbb59-2525-4ce7-a7fb-df03072756e1">